### ISSUE TITLE
Rework perma deletions for issues

### ISSUE THREAD

**Issue by nikitasius** [isaacs/github#1430]
Actually we can delete issues on github.

This is ruins **transparency** for **public** opensource projets.

I propose:
* add the button `Enable perma deletions for issues` with "no-way-to-go-back".
* **public** repositories where this feature enabled should have a sticker "Here maintainers can completely delete issues"

In private repositories people can do all what they want. Thats private.

**Comment by InDieTasten** [comment]
I agree that this could potentially become a problem, but for important repos, it would be very obvious if all of the sudden issues are missing. You can see missing IDs, and you can often check archive.org for the issues original content.

If you do not trust a maintainer, you sadly are out of luck. Deleting issues is just one thing. Maintainers can already do so much more than that.

Many "real" open source projects also have alternative bug-tracking software. Nobody has ever argued that they are not transparent, because they run their own bug tracker, even though I'm sure they can delete or even modify what's being posted on them.

For me at least, it's a nice-to-have

**Comment by nikitasius** [comment]
> If you do not trust a maintainer, you sadly are out of luck. Deleting issues is just one thing. Maintainers can already do so much more than that.

All is saved in a history. Also deleted comments should be accessible. There are a lot of shit with issues and censure. For example old Tox issues which became a drame with deletions and cleaning 3 years ago.

> Many "real" open source projects also have alternative bug-tracking software.
Why spend resources/time for this? I.e. there are different between windows10 -- freeBSD. I'm somewhere between then with my debian. About paranoid stuff - i don't have kernel repo or any what can shakeup the world, but there are another projects and every mainteiner have ambitions.
So mark what "this guy can delete issues" very useful. People love to do shit to each other in private situatuins and pissing off when it;s becomes public.

I don't need reddit here, github is not the reddit. Let people work normally, with transparent public repos. Or just show who isn't agree with.

Finally `alternative bug-tracking` goes to be alternative to github. Cause Redmond actually cooking the shitburger. They started with ToS and never stopped.

> For me at least, it's a nice-to-have
But realisation sucks. They (github) can do better.

**Comment by jbruchon** [comment]
LOL, not this nonsense again. We've been over this in the thread asking for the feature. You have made absolutely no compelling case for this notion that transparency is being harmed by this ability. Make a much better case than just your unsubstantiated belief.

**Comment by nikitasius** [comment]
> Make a much better case than just your unsubstantiated belief.

I will try to be clear: такого рода нововведения подрывают авторитет сего проекта, как opensource решения. Почему эту гадость ввели только сейчас? Все очень просто - после сделки с редмондом, они засрали ToS, следом начали и засирать сам сервис, переводя его на рельсы курса головной компании. Эта ужасная оплошность по стороны разработчиков гитхаба выйдет им же боком. В данной реализации она как козе баян, а Microsoft и Github это как как мартышка и очки. Хотя бы в задницу не засунули.

Btw, you can repeat and repeat again and again, i.e. copypaste works well.

**Comment by InDieTasten** [comment]
@nikitasius Clearly, there are multiple opinions on this. I can imagine a rework of the system where issues that do not violate Github ToS can not be hidden completely, but only "archived" or something along those lines. However that might be realised.

But to keep this thread alive, did you contact support and told them about your concerns, and if yes, could you share the outcome?

**Comment by nikitasius** [comment]
> But to keep this thread alive, did you contact support and told them about your concerns, and if yes, could you share the outcome?

Nope. I just let my issue here.

**Comment by erkinalp** [comment]
I want my issues and comments to be erased when I click delete.

**Comment by jbruchon** [comment]
@erkinalp Agreed, "delete" does not mean "archive." This notion that people should not be able to have control over their own repositories is stupid.

**Comment by nikitasius** [comment]
> @erkinalp Agreed, "delete" does not mean "archive." This notion that people should not be able to have control over their own repositories is stupid.

Technically you control nothing here:
> You grant us and our legal successors the right to store, parse, and display Your Content, and make incidental copies as necessary to render the Website and provide the Service

They are decide how to display it, or not :laughing:

**Comment by erkinalp** [comment]
Right to erasure is a statutory right, not a contractual one.
